### PR TITLE
Add callout for licensing in kubernetes docs

### PR DESCRIPTION
**Pull request by gabrieljackson**
This callout is intended to help individuals who will be using the Mattermost Operator, but don't have an enterprise license.

Fixes https://mattermost.atlassian.net/browse/CLD-8138

### CODE REVIEW

**Review by stylianosrigas — APPROVED** [review]
Thanks @gabrieljackson 🚀

File: source/install/install-kubernetes.rst [hunk 1]
```diff
@@ -107,6 +107,10 @@ Deploy Mattermost
     stringData:
       license: <LICENSE_FILE_CONTENTS>
 
+  .. note::
+    An enterprise license is required for a seamless experience running a Mattermost installation with multiple server instances. If you will be deploying Mattermost without an enterprise license it is recommended to only use a single server instance. This can be done by adding ``Replicas:  1`` to the ``spec`` section in step 2.
```
Comment (cwarnermm) [inline source/install/install-kubernetes.rst:111]:
```suggestion
    A Mattermost Enterprise license is required for a seamless experience deploying Mattermost with multiple server instances. If you plan to deploy Mattermost without an Enterprise license, we recommend a single server deployment by adding ``Replicas:  1`` to the ``spec`` section in step 2.
```

File: source/install/install-kubernetes.rst [hunk 2]
```diff
@@ -107,6 +107,10 @@ Deploy Mattermost
...
+    For more information a running highly-available Mattermost deployment please refer to :doc:`this documentation </scale/high-availability-cluster-based-deployment>`.
```
Comment (cwarnermm) [inline source/install/install-kubernetes.rst:112]:
```suggestion
    See the :doc:`high availability </scale/high-availability-cluster-based-deployment>` documentation for details on running a highly-available Mattermost deployment.
```

**Review by cwarnermm — APPROVED** [review]
Provided editorial feedback inline

**Comment by gabrieljackson** [comment]
> Provided editorial feedback inline

Thanks @cwarnermm! I appreciate the suggestions.

**Pull request merged**